python/pyarrow/tests/test_array.py

```diff
@@ -926,6 +931,16 @@ def test_cast_date32_to_int():
     assert result2.equals(arr)
 
 
+@pytest.mark.xfail(strict=True)  # TODO implement duration cast
```

Comment:
Just open a JIRA instead of adding a failing test.

Reply:
I already opened https://issues.apache.org/jira/browse/ARROW-6778 for that (but I also already wrote the test when I found out, so thought to just keep it)